core/src/main/scala/com/raphtory/internals/components/ingestion/IngestionExecutor.scala

```diff
@@ -14,55 +15,63 @@ import com.raphtory.internals.components.querymanager.NonBlocking
 import com.typesafe.scalalogging.Logger
 import org.slf4j.LoggerFactory
 
 import scala.concurrent.Future
 import scala.concurrent.duration.DurationInt
+import scala.concurrent.duration.FiniteDuration
 
-private[raphtory] class IngestionExecutor(
+private[raphtory] class IngestionExecutor[F[_]: Async](
     graphID: String,
     source: Source,
     blocking: Boolean,
     sourceID: Int,
     conf: Config,
-    topics: TopicRepository,
-    override val scheduler: Scheduler
-) extends Component[Any](conf)
-        with FlushToFlight {
-  override val logger: Logger                               = Logger(LoggerFactory.getLogger(this.getClass))
-  private val failOnError                                   = conf.getBoolean("raphtory.builders.failOnError")
-  override val writers: Map[Int, EndPoint[GraphAlteration]] = topics.graphUpdates(graphID).endPoint()
-  private val queryManager                                  = topics.blockingIngestion(graphID).endPoint
-  private val sourceInstance                                = source.buildSource(graphID, sourceID)
-  private val spoutReschedulesCount                         = telemetry.spoutReschedules.labels(graphID)
-  private val fileLinesSent                                 = telemetry.fileLinesSent.labels(graphID)
-
-  private var index: Long                              = 0
-  private var scheduledRun: Option[() => Future[Unit]] = None
+    topics: TopicRepository
+) extends TelemetryReporter {
+  private val logger: Logger                               = Logger(LoggerFactory.getLogger(this.getClass))
+  private val failOnError                                  = conf.getBoolean("raphtory.builders.failOnError")
+  private val writers: Map[Int, EndPoint[GraphAlteration]] = topics.graphUpdates(graphID).endPoint()
+  private val queryManager                                 = topics.blockingIngestion(graphID).endPoint
+  private val sourceInstance                               = source.buildSource(graphID, sourceID)
+  private val spoutReschedulesCount                        = telemetry.spoutReschedules.labels(graphID)
+  private val fileLinesSent                                = telemetry.fileLinesSent.labels(graphID)
+
+  private var index: Long = 0
 
   sourceInstance.setupStreamIngestion(writers)
 
-  private def rescheduler(): Unit = {
-    sourceInstance.executeReschedule()
-    executeSpout()
-  }: Unit
-
-  override def stop(): Unit = {
-    scheduledRun.foreach(cancelable => cancelable())
-    close()
-    writers.values.foreach(_.close())
-  }
-
-  override def run(): Unit = {
-    logger.debug("Running ingestion executor")
-    executeSpout()
-  }
-
-  override def handleMessage(msg: Any): Unit = {} //No messages received by this component
-
-  private def executeSpout(): Unit = {
+  def release(): F[Unit] =
+    for {
+//      _ <- Async[F].delay(close()) -> Needed if this class extends FlushToFlight
+      _ <- Async[F].delay(writers.values.foreach(_.close()))
```

Comment:
are any of the calls wrapped by `delay` blocking?
if so we should use `blocking`

Reply:
The `close` implementation of the `AkkaConnector` is indeed empty. I'll bear that in mind though when reimplementing the writer interface